### PR TITLE
Testing music genre classes

### PR DESCRIPTION
**Pull request by braincee**
_**In this pull request, the following requirements were implemented:**_

 - I created the `Music` Album class in a `separate` rb file.

 - I created a `Genre` class with an association to the Item class (in a separate .rb file).

 All Music Album class properties visible in the diagram should be defined and set up in the constructor method.

 All Genre class properties visible in the diagram should be defined and set up in the constructor method.

 Implement methods I did :

 - add_item method in the `Genre` class
 - should take an instance of the Item class as an input
 - should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)
 - can_be_archived?() in the `Music` Album class
 - should override the method from the parent class
 - should return true if the parent's method returns true AND if on_spotify equals true
    otherwise, it should return false
 - I added unit tests for all implemented methods.

 The following options should be available:
 - a list of all music albums
 - a list of all genres (e.g 'Comedy', 'Thriller')
 - added a music album
 - preserved all data preserved by saving collections in .json files.
 - created a `schema` .sql file with tables that will be analogical to the structure of the classes that I  created:
 - music_albums table (add all properties and associations from the parent Item class as table columns)
 - genres table

### CODE REVIEW

File: spec/genre_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  context 'When retrieving the name of a genre' do
+    it 'should return the genre name' do
+      expect(@genre.name).to eq('Reggae')
+    end
+  end
+
+  context 'When adding a music album into a genre' do
+    it 'should add the music album to the genre' do
+      expect(@genre.add_item(@music_album)).to include(@music_album)
+    end
```
Comment (Baroka-wp) [inline spec/genre_spec.rb:26]: ## Good test 👍

**Review by Baroka-wp — COMMENTED** [review]
## Thank you @braincee for your hard work !
### ## Highlights ♨️ 
- No Linter error ✔️ 
- Good and descriptive PR 💯 
- Correct github flow

Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Review by KanoCode — APPROVED** [review]
GitHub Flow was followed ✔️
No linter errors ✔️
Descriptive pull request ✔️
Descriptive commit messages ✔️
Good code organization and working app ✔️
Descriptive README ✔️
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Comment by braincee** [comment]
Thanks very much for the review @KanoCode

**Comment by braincee** [comment]
Thanks very much, @Baroka-wp and @KanoCode 

thank-curlscakes

**Pull request merged**